whyis/database/elasticstore.py

```diff
@@ -0,0 +1,469 @@
+from builtins import str
+from builtins import range
+from six import b
+from six.moves.urllib.request import pathname2url
+from six import iteritems
+
+from rdflib.store import Store, VALID_STORE, NO_STORE
+from rdflib.term import URIRef, BNode
+
+from rdflib.graph import Graph
+
+import requests
+from uuid import uuid4
+from urllib.parse import quote_plus
+
+import rdflib
+
+import json
+
+import logging
+logger = logging.getLogger(__name__)
+
+__all__ = ['ElasticSearchStore']
+import pkg_resources
+
+elastic_index_settings = pkg_resources.resource_string(__name__, "elastic_index_settings.json")
+
+#from rdflib.plugins.sparql import CUSTOM_EVALS
+from rdflib.plugins.sparql.sparql import Query, AlreadyBound
+
+from rdflib.plugins.sparql.parser import parseQuery, parseUpdate
+from rdflib.plugins.sparql.algebra import translateQuery, translateUpdate
+
+from rdflib.plugins.sparql.evalutils import (
+    _filter, _eval, _join, _diff, _minus, _fillTemplate, _ebv, _val)
+
+#from rdflib.plugins.sparql.evaluate import evalPart
+from elastic_eval import evalPart
+from rdflib.plugins.sparql.evaluate import evalBGP as evalBGP_orig
+from rdflib.plugins.sparql.evaluate import evalGraph as evalGraph_orig
+
+class ElasticSearchStore(Store):
+    context_aware = True
+    formula_aware = False
+    transaction_aware = False
+    graph_aware = True
+    __prefix = {}
+    __namespace = {}
+
+    def __init__(self, configuration=None, identifier=None):
+        self.__open = False
+        self.__identifier = identifier
+        super(ElasticSearchStore, self).__init__(configuration)
```

Comment:
All members should be set to dummy values in __init__, such as self.__session = None